### PR TITLE
HW-259: Styling new mailing page including some other basic components

### PR DESCRIPTION
**Pull request by meladawy**
Before: 
before

After:
download

### CODE REVIEW

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 1]
```diff
@@ -12,25 +12,167 @@
     top: 2px;
 }
 
-.form-control.select2-container {
+.form-control.select2-container,
+.form-group .select2-container {
     padding: 0;
     height: auto;
+    background: white;
```
Comment (meladawy) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: Aaah sorry i'll change this

File: scss/bootstrap/overrides/style/_wizard.scss [hunk 2]
```diff
@@ -0,0 +1,114 @@
+.crmb-wizard {
+  .crmb-wizard-title,
```
Comment (boriszaydlin) [inline scss/bootstrap/overrides/style/_wizard.scss]: Add an empty line here

File: scss/bootstrap/overrides/style/_wizard.scss [hunk 3]
```diff
@@ -0,0 +1,114 @@
+.crmb-wizard {
```
Comment (boriszaydlin) [inline scss/bootstrap/overrides/style/_wizard.scss:1]: Add an empty line here

File: scss/bootstrap/overrides/style/_wizard.scss [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      }
+
+      &:hover {
+        background: none;
+        color: $brand-primary;
+      }
+    }
+  }
+
+  .crmb-wizard-title .crmb-wizard-number {
```
Comment (boriszaydlin) [inline scss/bootstrap/overrides/style/_wizard.scss]: You can use
```
.crmb-wizard {
  &-title { // styles }
  &-body .form-control { // styles }
  &-body .form-group { // styles }
  &-body .select2-container { // styles }
}
```

File: scss/bootstrap/overrides/style/_wizard.scss [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    width: 32px;
+    height: 32px;
+    margin-right: 5px;
+    border: 2px solid;
+    border-radius: 50px;
+    text-align: center;
+    display: inline-block;
+  }
+
+  .crmb-wizard-body .form-group {
```
Comment (boriszaydlin) [inline scss/bootstrap/overrides/style/_wizard.scss]: For this and below see here: https://github.com/compucorp/org.civicrm.shoreditch/pull/1/files#r107909462
Or you can use in such way:
```
.crmb-wizard-body {
  .form-group {}
  .form-control {}
  .crmb-wizard-body .form-group .select2-container {}
  .select2-container {}
}
```

**Comment by jamienovick** [comment]
Hi Maged - looks good! Just wondering about grid width. Should we have the white bit of the page wider with less border? Can we make the fields a bit wider also perhaps as the title and the groups may be a long list? Thanks

**Comment by jamienovick** [comment]
ps. Enable campaigns in civicrm enabled components to check that element is styled also! (civicrm/admin/setting/component?reset=1)

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 6]
```diff
@@ -12,25 +12,167 @@
...
 
-.form-control .select2-container.select2-dropdown-open .select2-choices {
+.form-control .select2-container.select2-dropdown-open .select2-choices,
+.form-group .select2-container.select2-dropdown-open .select2-choices {
     border-radius: 3px 3px 0 0;
 }
 
-.form-control.select2-container .select2-choices {
+.form-control.select2-container .select2-choices,
+ul.select2-choices {
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: Please use 2 spaces for left indent

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 7]
```diff
@@ -12,25 +12,167 @@
...
+      font-size: 1em;
+
+      &:after {
+        @include fa-icon($content: '\f00d', $color: $crm-white);
+      }
+    }
+}
+
+.form-control.select2-container .select2-choice,
+.form-group .select2-container .select2-choice {
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: Same as here:
https://github.com/compucorp/org.civicrm.shoreditch/pull/1/files#r107910732

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 8]
```diff
@@ -12,25 +12,167 @@
...
+    line-height: 30px;
+    border: none;
+    
+    .select2-arrow {
+      background: none;
+    }
+}
+
+.form-control.select2-container .select2-input,
+.form-group .select2-container .select2-input {
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: and here
https://github.com/compucorp/org.civicrm.shoreditch/pull/1/files#r107910732

File: scss/civicrm/common/_fields.scss [hunk 9]
```diff
@@ -0,0 +1,7 @@
+.crm-mailing-recipients-row {
```
Comment (boriszaydlin) [inline scss/civicrm/common/_fields.scss:1]: Add an empty line here

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 10]
```diff
@@ -12,25 +12,167 @@
...
+
+    li {
+      padding: 5px;
+      background: $crm-white;
+    }
+
+    .select2-result-label {
+      padding: 5px 10px;
+      color: $gray-dark;
+
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: According to SMACSS we should not use more than three level nesting
See below too

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 11]
```diff
@@ -12,25 +12,167 @@
...
+          }
+        }
+      }
+    }
+
+    .select2-highlighted {
+      background: $crm-background;
+      color: $gray-dark;
+
+      .crm-select2-row-description {
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss:171]: empty line

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 12]
```diff
@@ -12,25 +12,167 @@
...
+        text-align: center;
+
+        &:before {
+          font-size: $font-size-base;
+          position: relative;
+          bottom: 2px;
+        }
+
+        &.Individual-icon {
+          background: none;
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: empty line

File: scss/bootstrap/mixins/_fa-icon.scss [hunk 13]
```diff
@@ -16,3 +16,22 @@
     color: $gray-dark;
   }
 }
+
+@mixin fa-icon2($font-size: null, $content: null, $color: null) {
```
Comment (boriszaydlin) [inline scss/bootstrap/mixins/_fa-icon.scss]: Why you made a copy of
`@mixin fa-icon` ?

File: scss/bootstrap/overrides/style/_wizard.scss [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+    li:not(.active) a {
+      color: $gray-darker;
+
+      &:after {
+        content: "";
+        position: absolute;
+        bottom: 0;
+        right: -9px;
+        width: 1px;
+        height: 31px;
```
Comment (boriszaydlin) [inline scss/bootstrap/overrides/style/_wizard.scss]: according to http://caniuse.com/#search=transform
we should use transform with prefixes

File: scss/bootstrap/overrides/style/_wizard.scss [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+      }
+
+      &:before {
+        content: "";
+        position: absolute;
+        top: 0;
+        right: -9px;
+        width: 1px;
+        height: 31px;
+        transform: skewX(15deg);
```
Comment (boriszaydlin) [inline scss/bootstrap/overrides/style/_wizard.scss]: same as here: https://github.com/compucorp/org.civicrm.shoreditch/pull/1/files#r107913198

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 16]
```diff
@@ -12,25 +12,167 @@
...
     height: auto;
+    background: white;
+    border: 1px solid $crm-darker-grayblue;
+    box-shadow: $crm-form-control-shadow;
+    height: 30px;
 }
 
-.form-control.select2-container.select2-dropdown-open {
+.form-control.select2-container.select2-dropdown-open,
+.form-group .select2-container.select2-dropdown-open {
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: Are you sure using color without variable?

File: scss/bootstrap/overrides/style/_wizard.scss [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+      padding: 15px 20px;
+      color: $brand-primary;
+    }
+
+    li.active a {
+      background: $crm-light-gray;
+      position: relative;
+      color: $brand-primary;
+
+      &:after {
```
Comment (boriszaydlin) [inline scss/bootstrap/overrides/style/_wizard.scss]: Same as here:
https://github.com/compucorp/org.civicrm.shoreditch/pull/1/files#r107911442

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 18]
```diff
@@ -12,25 +12,167 @@
...
+
+        &.Individual-icon {
+          background: none;
+          &:before {
+            @include fa-icon($content: "\f007");
+          }
+        }
+
+        &.Organization-icon {
+          background: none;
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: empty line

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 19]
```diff
@@ -12,25 +12,167 @@
...
+          &:before {
+            @include fa-icon($content: "\f0f7");
+          }
+        }
+
+        &.Household-icon {
+          background: none;
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: empty line

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 20]
```diff
@@ -12,25 +12,167 @@
...
+      right: 2px;
+      left: auto;
+      color: $crm-white;
+      background: $brand-primary;
+      top: 50%;
+      transform: translateY(-50%);
+      line-height: 10px;
+      text-align: center;
+      box-sizing: border-box;
+      border-radius: 25px;
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: Do we need variable here? (for font size)

Comment (meladawy) [reply scss/bootstrap/vendor/select2/select2-bootstrap.scss]: I believe we should use variables with pixel size based fonts. but with em its okay to have it like this

**Review by meladawy — COMMENTED** [review]
@boriszaydlin i fixed most of ur comments plz review. 


> Should we have the white bit of the page wider with less border?

@jamienovick not sure if fully understand this part of your comments (can u explain more). I also did some changes to field width

screen shot 2017-03-24 at 7 08 06 pm

File: scss/bootstrap/overrides/style/_wizard.scss (hunk 3 above)
Comment (meladawy) [reply scss/bootstrap/overrides/style/_wizard.scss:1]: No need for this line (as we didn't find any reference for such standard)

File: scss/civicrm/common/_fields.scss (hunk 9 above)
Comment (meladawy) [reply scss/civicrm/common/_fields.scss:1]: No need to add empty line here as we didn't find any guide for this and most of current cases doesn't apply an empty line in such case

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss (hunk 11 above)
Comment (meladawy) [reply scss/bootstrap/vendor/select2/select2-bootstrap.scss:171]: I believe we shouln't add new line in such case

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 21]
```diff
@@ -1,94 +1,257 @@
...
+  margin-bottom: 0;
+  background: none;
+
+  .select2-search-choice {
+    background: $crm-white;
+    border-color: $brand-primary;
+    padding: 5px 20px 5px 5px;
+  }
+
+  .select2-search-choice .select2-search-choice-close {
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss:78]: please sort by alphabet for all styles

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 22]
```diff
@@ -1,94 +1,257 @@
...
+    right: 2px;
+    left: auto;
+    color: $crm-white;
+    background: $brand-primary;
+    top: 50%;
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: use mixin for prefixes

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 23]
```diff
@@ -1,94 +1,257 @@
...
-    border-color: #5897FB;
-    border-radius: 3px 3px 0 0;
+.form-control.select2-container.select2-dropdown-open,
+.form-group .select2-container.select2-dropdown-open {
+  border-color: $crm-select2-dropdown-border;
+  border-radius: 3px 3px 0 0;
 }
 
-.form-control .select2-container.select2-dropdown-open .select2-choices {
-    border-radius: 3px 3px 0 0;
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: Are you sure about space between .form-control and .select2-container? I see there is no space in below and above code

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 24]
```diff
@@ -1,94 +1,257 @@
 .form-control .select2-choice {
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: don't we need here this selector? .form-group .select2-container

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 25]
```diff
@@ -1,94 +1,257 @@
...
-    border: 0;
-    border-radius: 2px;
-    text-decoration: none !important;
+  border: 0;
+  border-radius: 2px;
+  text-decoration: none !important;
 }
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: and here? .form-group .select2-container

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 26]
```diff
@@ -1,94 +1,257 @@
...
+  border-radius: 0 2px 2px 0;
+  background: transparent;
+  border-left: 0 none;
+  right: 4px;
+  top: 2px;
 }
 
-.form-control.select2-container {
-    padding: 0;
-    height: auto;
```
Comment (boriszaydlin) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss:17]: I believe we can make those selctors:
.form-control.select2-container,
.form-group .select2-container { }
as main and than use nesting

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss (hunk 25 above)
Comment (meladawy) [reply scss/bootstrap/vendor/select2/select2-bootstrap.scss]: No its okay to leave it like this .form-group wrapping .select2-container

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss (hunk 26 above)
Comment (meladawy) [reply scss/bootstrap/vendor/select2/select2-bootstrap.scss:17]: This is resolved

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss (hunk 21 above)
Comment (meladawy) [reply scss/bootstrap/vendor/select2/select2-bootstrap.scss:78]: Done

File: scss/bootstrap/vendor/select2/select2-bootstrap.scss [hunk 27]
```diff
@@ -1,94 +1,250 @@
...
+  box-shadow: $crm-form-control-shadow;
+  height: 30px;
+  padding: 0;
+
+  .select2-choice {
     border: 0;
     border-radius: 2px;
-    text-decoration: none !important;
-}
+    text-decoration: none!important;
```
Comment (meladawy) [inline scss/bootstrap/vendor/select2/select2-bootstrap.scss]: @boriszaydlin 
Not sure why the beautifier removed the space between !important and none. what is the correct format?

File: scss/bootstrap/overrides/style/_wizard.scss [hunk 28]
```diff
@@ -0,0 +1,118 @@
...
+        padding: 15px 20px;
+        color: $brand-primary;
+      }
+
+      li.active a {
+        background: $crm-light-gray;
+        position: relative;
+        color: $brand-primary;
+
+        &:after {
```
Comment (iVolodya) [inline scss/bootstrap/overrides/style/_wizard.scss]: We can combine same styles

```
&:before,
&:after {
  content: "";
  position: absolute;
  top: 0;
  bottom: 0;
  width: 0;
  height: 100%;
  border-color: transparent;
  border-style: solid;
  border-width: 30px 0px 30px 8px;
}
```

Comment (BorisCompucorp) [inline scss/bootstrap/overrides/style/_wizard.scss]: We can combine those lines:

```
&:after,
&:before {
  content: "";
  position: absolute;
  top: 0;
  bottom: 0;
  width: 0;
  height: 100%;
  border-color: transparent;
  border-style: solid;
  border-width: 30px 0px 30px 8px;
}
```

File: scss/bootstrap/overrides/style/_wizard.scss [hunk 29]
```diff
@@ -0,0 +1,118 @@
...
+      }
+
+      li.active:first-child a:before {
+        display: none;
+      }
+
+      li:not(.active) a {
+        color: $gray-darker;
+
+        &:after {
```
Comment (BorisCompucorp) [inline scss/bootstrap/overrides/style/_wizard.scss]: We can combine this same styles too:

```
&:after,
&:before {
  @include prefix(transform, skewX(-15deg), webkit ms moz);
  content: "";
  position: absolute;
  right: -9px;
  width: 1px;
  height: 31px;
  background: $crm-light-gray;
}
```

**Review by BorisCompucorp — APPROVED** [review]

**Pull request merged**